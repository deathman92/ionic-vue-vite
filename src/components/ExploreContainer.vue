<template>
  <IonCard class="ion-text-center ion-padding">
    <IonTitle size="large">{{ name }}</IonTitle>
    <p>
      Explore
      <a
        target="_blank"
        rel="noopener noreferrer"
        href="https://ionicframework.com/docs/components"
      >
        UI Components
      </a>
    </p>
  </IonCard>
</template>

<script setup lang="ts">
defineProps<{
  name: string;
}>();
</script>
