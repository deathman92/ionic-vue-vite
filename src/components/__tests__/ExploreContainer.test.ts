import { render } from "@testing-library/vue";

import ExploreContainer from "../ExploreContainer.vue";

describe("ExploreContainer", () => {
  it("renders correctly", () => {
    const wrapper = render(ExploreContainer, { props: { name: "Test Name" } });

    expect(wrapper).toBeTruthy();

    expect(wrapper.queryByText("Test Name")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const wrapper = render(ExploreContainer, { props: { name: "Test Name" } });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
