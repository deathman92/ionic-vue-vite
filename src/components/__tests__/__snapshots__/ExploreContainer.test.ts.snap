// Vitest Snapshot v1

exports[`ExploreContainer > matches snapshot 1`] = `
<ion-card class="ion-text-center ion-padding">
  <ion-title class>Test Name</ion-title>
  <p> Explore <a href="https://ionicframework.com/docs/components" rel="noopener noreferrer" target="_blank"> UI Components </a>
  </p>
</ion-card>
`;
